kernel/term_typing.ml

```diff
@@ -267,6 +267,8 @@ let infer_declaration (type a) ~(trust : a trust) env (dcl : a constant_entry) =
       let tyj = infer_type env typ in
       let proofterm =
         Future.chain body (fun ((body,uctx),side_eff) ->
+          (* don't redeclare universes which are declared for the type *)
+          let uctx = Univ.ContextSet.diff uctx univs in
```

Comment:
Not sure if this is the right place to do this, but it seemed cleaner than doing it in the proof_global function and it makes OK sense to allow the redundancy in the entry.